crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -262,6 +265,15 @@ fn heroes_filter<'a>(
     move |user_id| user_id != own_user_id && !member_hints.service_members.contains(user_id)
 }
 
+/// The kind of room member updates that just happened.
+#[derive(Debug, Clone)]
+pub enum RoomMembersUpdateKind {
+    /// The whole list room members was reloaded.
+    FullReload,
+    /// A few members were updated, their user ids are included.
+    Partial(BTreeSet<OwnedUserId>),
```

Comment:
While this distinction is not used at the moment, I hope it'll make building a live room members observable in the future a bit easier.